plugins/org.jkiss.dbeaver.net.ssh/src/org/jkiss/dbeaver/model/net/ssh/SSHUtils.java

```diff
@@ -78,4 +90,94 @@ public static boolean isKeyEncrypted(byte[] privKeyValue) {
+    private static final String KNOWN_SSH_HOSTS_FILE_NAME = "known_hosts";
+
+    @NotNull
+    public static File getKnownSshHostsFileOrDefault() {
+        return  getKnownSshHostsFileImpl(true);
+    }
+
+    @Nullable
+    public static File getKnownSshHostsFileOrNull() {
+        return  getKnownSshHostsFileImpl(false);
+    }
+
+    private static File getKnownSshHostsFileImpl(boolean forceFileObjectOnFail) {
+        String sshHomePathString = Platform.getPreferencesService().getString(PLATFORM_SSH_PREFERENCES_NODE, PLATFORM_SSH_PREFERENCES_SSH2HOME_KEY, null, null);
+        if (CommonUtils.isNotEmpty(sshHomePathString)) {
+            try {
+                return Paths.get(sshHomePathString, KNOWN_SSH_HOSTS_FILE_NAME).toFile();
+            } catch (InvalidPathException e) {
+                log.warn("Failed to resolve SSH known hosts file location at " + sshHomePathString, e);
+                if (forceFileObjectOnFail) {
+                    return new File(sshHomePathString + File.pathSeparator + KNOWN_SSH_HOSTS_FILE_NAME);
+                } else {
+                    return null;
+                }
+            }
+        } else {
+            // seems preference path not set at all, so try to resolve it and preserve
+            return resolveDefaultKnownSshHostsFile(forceFileObjectOnFail, true);
+        }
+    }
+
+    private static File resolveDefaultKnownSshHostsFile(boolean forceFileObjectOnFail, boolean updatePreferences) {
+        try {
+            String userHomePathString = System.getProperty(IConstants.SYSTEM_PROPERTY_USER_HOME);
+            if (userHomePathString != null) {
+                Path userHomeDirPath = Paths.get(userHomePathString);
+                if (userHomeDirPath.toFile().isDirectory()) {
+                    Path sshHomeDirPath = userHomeDirPath.resolve(DEFAULT_SSH_HOME_DIR_NAME);
+                    File sshHomeDir = sshHomeDirPath.toFile();
+
+                    if (RuntimeUtils.isWindows() && (!sshHomeDir.isDirectory() || !sshHomeDir.exists())) {
+                        Path sshHomeOldDirPath = userHomeDirPath.resolve(DEFAULT_SSH_HOME_DIR_NAME_WIN_OLD);
+                        File sshHomeOldDir = sshHomeDirPath.toFile();
+                        if (sshHomeOldDir.isDirectory()) {
+                            sshHomeDirPath = sshHomeOldDirPath;
+                            sshHomeDir = sshHomeOldDir;
+                        }
+                    }
+
+                    if (sshHomeDir.isDirectory() || !sshHomeDir.exists()) {
+                        // don't need to create it until we'll need to write known hosts file
+
+                        if (updatePreferences) {
+                            Platform.getPreferencesService()
+                                    .getRootNode().node(PLATFORM_SSH_PREFERENCES_NODE)
+                                    .put(PLATFORM_SSH_PREFERENCES_SSH2HOME_KEY, sshHomeDir.getAbsolutePath());
```

Comment:
We use 4 spaces for continuation indentation.

```suggestion
                                .getRootNode().node(PLATFORM_SSH_PREFERENCES_NODE)
                                .put(PLATFORM_SSH_PREFERENCES_SSH2HOME_KEY, sshHomeDir.getAbsolutePath());
```

In case of Intellij IDEA, code style can be adjusted in <kbd>Editor</kbd> &rArr; <kbd>Code Style</kbd> &rArr; <kbd>Java</kbd> &rArr; <kbd>Continuation indent</kbd>.

Reply:
ok, thanks